### ISSUE TITLE
Actions in sidebar when a search is active should act upon only visible tabs rather than all

### ISSUE THREAD

**Issue by tokumeiii** [josh-berry/tab-stash#399]
### Problem Statement

Prior to version 3.0, setting a search filter in Tab Stash's sidebar view would cause the "stash all open tabs to a new group" button to stash only the tabs that are visible in the current filtered view. As of 3.0, it now stashes all tabs in the window including the ones that are hidden from the sidebar view. I think that it may have worked the same way for the "close all unstashed/stashed tabs" buttons, but I didn't frequently use those (because I have the extension set to automatically close stashed tabs) so I don't remember for certain.

I'm not entirely sure if this change in behavior is intended or not, since I do acknowledge that the tooltips for the buttons do not imply that the filtering will change their behavior. I found it very useful, however; almost all of my interaction with Tab Stash up to this point has involved opening the sidebar, setting a filter, and then stashing all of the tabs that matched the filter with that one click of the "stash all open tabs" button.

### Preferred Solution(s)

I'd like to see the buttons in the sidebar view in general only act upon tabs that are visible in the current filtered view. The buttons to stash tabs from the open window, close tabs in the open window, or restore tabs from a saved group to the open window. To me, it seems like since filtering is an explicit action by the user to focus in on a subset the tabs they're looking at, it stands to reason that any actions they then initiate from that view are probably mentally intended for that subset of tabs rather than all of the tabs in the window.

### Alternative Solution(s)

Alternatively, at least having some sort other way to narrow the focus (e.g. holding down a key while clicking the button) would be appreciated.

### Additional Context

<hr>
<em>Readers: If you are also interested in seeing this feature be developed, please vote for it by giving the ORIGINAL POST a thumbs-up using the :smiley: button.  You are welcome to leave comments and discuss the feature request, but "Me too!" comments are not counted by the voting system.</em>

**Comment by josh-berry** [comment]
Yeah, this was definitely a bug (which I never noticed) in the older versions, but I completely understand wanting the buttons to only act on visible things. I'll give some thought to how to address this, thanks for letting me know.

In the meantime, you can Shift+Click to select only the visible items and then stash/close them. Hope that helps!
